File: acmicpc/acmicpc11047.py
import time
start = time.time()

import sys

N, K = map(int, sys.stdin.readline().split())
arr = []
for _ in range(N):
    arr.append(int(sys.stdin.readline()))


t = []
for i in range(N-1, -1, -1):
    t.append(K // arr[i])
    K -= t[-1] * arr[i]

print(sum(t))

end = time.time()
print('elapsed time: {}'.format(end - start))

File: acmicpc/acmicpc11047.cpp
#include <iostream>

int main()
{
    int N, K;
    scanf("%d %d", &N, &K);
    int arr[N];
    
    for (int i=0; i < N; i++){
        scanf("%d", &arr[i]);
    }
    int t[N];
    for (int i=N-1; i >= 0; i--){
        t[i] = int(K / arr[i]);
        K = K - t[i] * arr[i];
    }
    int sum = 0;
    for (int i=0; i < N; i++){
        sum = sum + t[i];
    }
    printf("%d\n", sum);
}